src/opencl_common.c

```diff
@@ -1268,6 +1268,26 @@ static char *include_source(const char *pathname, int sequential_id, const char
 		full_path = quote_str(full_path);
 	}
 
+#ifdef __CYGWIN__
+	/*
+	 * Workaround for OpenCL kernels not found when run from Windows PowerShell.
+	 * CygWin (at least in PowerShell) maps the main JtR folder to "/".
+	 */
+	const char *sandbox_path;
+
+	sandbox_path = realpath(".", NULL);
+
+	if (sandbox_path && sandbox_path[0] == '/' && !strcmp(full_path, "/run/opencl")) {
+		int main_folder_fd;
+
+		if ((main_folder_fd = open("/run/opencl", O_DIRECTORY)) > 0) {
+			fchdir(main_folder_fd);
+			close(main_folder_fd);
+		}
+	}
+	MEM_FREE(sandbox_path);
+#endif
+
 	snprintf(include, LINE_BUFFER_SIZE,
 	         "-I %s %s %s%s%s%s%d %s%d %s -D_OPENCL_COMPILER %s",
 	        full_path,
```

Comment:
I find it weird that even after `chdir()` to the kernels directory we still pass `/run/opencl` as the include path. My guess was that the OpenCL backend is unaware of Cygwin's mappings, which is why it failed for include paths, but then it'd also be failing for `/run/opencl` even when we're in that directory. I expected we'd need to `chdir("/run");` and pass `-I opencl` here, or when we use `chdir("/run/opencl");` then pass `-I .` here.

Reply:
In that respect, the patch does nothing. It is the legacy code (which applies to other platforms and in the CMD). I don't need to add a `-I`, so I don't.

Reply:
I never implied we needed to _add_ a `-I`; I thought that we'd need to use a different argument to the existing `-I` after having used the `chdir()` trick.